File: 0370-Range Addition/0370-Range Addition.cpp
class Solution {
public:
    vector<int> getModifiedArray(int length, vector<vector<int>>& updates) {
        if (length <= 0)
        {
            return vector<int>();
        }
        
        vector<int> result(length);
        for (auto& update : updates)
        {
            result[update[0]] += update[2];
            if (update[1] < length - 1)
            {
                result[update[1] + 1] -= update[2];
            }
        }
        
        for (int i = 1; i < length; ++i)
        {
            result[i] += result[i - 1];
        }
        
        return result;
    }
};

File: 0301-0400/0370-Range Addition/0370-Range Addition.py
class Solution:
    def getModifiedArray(self, length: int, updates: List[List[int]]) -> List[int]:
        result = [0] * length
        for start, end, inc in updates:
            result[start] += inc
            if end + 1 < length:
                result[end + 1] -= inc
        for i in range(1, length):
            result[i] += result[i - 1]
        return result
